src/config.ts

```diff
@@ -38,9 +38,9 @@ export interface SwaggerConfig {
   entryFile: string;
 
   /**
-   * Set this to true if you want to prevent requests from coming in with more properties than are known/allowed
+   * Modes that allow you to prevent input data from entering into your API. This will document your decision in the swagger.yaml and it will turn on excess-property validation (at runtime) in your routes.
    */
-  noImplicitAdditionalProperties?: boolean;
+  noImplicitAdditionalProperties?: 'throw-on-extras' | 'silently-remove-extras';
```

Comment:
Can we add another flag like `trimAdditionalProperties`?
This would make sure there are no breaking changes.
If someone sets both to true, we can detect, warn and use the stricter `noImplicitAdditionalProperties`.
This reminds me of the various feature flags TS itself has. While it surely is not ideal, it punts the breaking change to 3.0, where we can consolidate both flags into on. What do you think @dgreene1?

Reply:
That was my original thought, but I really don’t like the idea that “true” would default to one of the two options and that it wouldn’t be clear to the user which one we’ve selected for them.

I don’t have the answer here. I’m kinda lost.

Reply:
@WoH I just spoke to one of the architects at my day job and they said that in their option the default (ie the value true relates to) should be the silently remove option. This leads me to think that if I were to default true to something, that’s what I would choose. However, that would also be a breaking change since when I released yesterday that first introduced this only had the throw option.

Reply:
I guess I still prefer warning users of the new version about invalid config (both flags true) over warning users impacted that their config is invalid (with the 2 flag case someone is actively setting something and presumably paying attention), although the recency of the change definitely makes it a lot more reasonable to go the other way.